atomdb/species.py

```diff
@@ -166,7 +154,9 @@ def __init__(self, x, y, log=False):
         self._log = log
         self._obj = CubicSpline(
             x,
-            np.log(y) if log else y,
+            # Clip y values to >=ε^2 if using log because they have to be above 0;
+            # having them be at least ε^2 seems to work based on my testing
+            np.log(y.clip(min=np.finfo(float).eps ** 2)) if log else y,
```

Comment:
@PaulWAyers and @msricher  I have a doubt, for all the features that we describe as splines, for values of `r` outside our array of values, the spline is a polynomial that will diverge for $r \to \infty$. If we know the correct decay for some of the features, why not use a switch function (like hyperbolic tangent) to change between the spline and the function (or even a constant value) outside its range?